### DISCUSSION TITLE
Pillow source

### DISCUSSION THREAD

**Discussion by Gi0rgi0s** [keithrozario/Klayers#404] [Q&A]
Hi all, where are the modules being pulled from? I am specifically looking for the pillow module. I am having difficulty downloading a working one using pip, but the one in the layer works great. Thanks!

**Comment by keithrozario** [comment]
This is purely just pip install from PyPI. Nothing fancy :)

**Comment by saintzst** [comment]
Facing the same issue. I've tried a pip install and made a layer myself but it didn't work. My function keeps throwing the error 

`Unable to import module 'lambda_function': cannot import name '_imaging' from 'PIL' (/opt/python/PIL/__init__.py)` 

Nowhere in my code import `_imaging`. I also tried installing via a wheel file, which didn't work either, while the layer in this repo works very well. I'm genuinely curious how it is different.

My function runtime is **python3.9.**

**Reply by keithrozario** [reply]
PIL tries to import _imaging, and that fails.

Are you building this in an Amazon Linux environment?

**Reply by curtinmjc** [reply]
I'm getting the same result as saintzst above with a Lambda function that has a "Specify an ARN" layer that uses "arn:aws:lambda:us-east-1:770693421928:layer:Klayers-p312-Pillow:3".  The Lambda function code is "from PIL import Image".
